framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -337,7 +337,43 @@ IterationAdaptiveDT::limitDTByFunction(Real & limitedDT)
 {
   Real orig_dt = limitedDT;
 
-  if (_timestep_limiting_function)
+  // Limit by function change for piecewise linear functions.
+  if (_piecewise_timestep_limiting_function && _max_function_change > 0)
+  {
+    auto linear_function = dynamic_cast<PiecewiseLinear *>(_piecewise_timestep_limiting_function);
+    if (linear_function)
+    {
+      const auto current_function_value = linear_function->value(_time, {});
+
+      for (auto next_time_index = 1u; next_time_index < _times.size(); ++next_time_index)
```

Comment:
1u is not the right type. While we are unlikely to exceed the range here (really unlikely) we prefer matching the type when possible. Two options:
1) std::size_t `next_time_index`
2) MooseIndex(_times) `next_time_index`

Reply:
done